scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -133,4 +229,29 @@ private void extractAnalyzerImportsAndDependencies(ScanTomlFile.Analyzer analyze
     private void buildStringWithNewLine(StringBuilder stringBuilder, String content) {
         stringBuilder.append(content).append(System.lineSeparator());
     }
+
+    List<Issue> runExternalAnalyzers(Project project, Map<String, List<Rule>> externalAnalyzers) {
+        List<ScannerContext> scannerContextList = new ArrayList<>();
+        for (Map.Entry<String, List<Rule>> externalAnalyzer : externalAnalyzers.entrySet()) {
+            ScannerContextImpl scannerContext = new ScannerContextImpl(externalAnalyzer.getValue());
+            scannerContextList.add(scannerContext);
+            Map<String, Object> pluginProperties = new HashMap<>();
+            pluginProperties.put("ScannerContext", scannerContext);
+            project.projectEnvironmentContext().getService(CompilerPluginCache.class)
+                    .putData(externalAnalyzer.getKey(), pluginProperties);
```

Comment:
Are we populating this map for some side effect? May be put a comment if so

Reply:
Yes, the static analyzer compiler plugins retrieve the context passed from this map via the `CompilerPluginContext` and perform additional analysis when they get engaged during project compilation